google/auth/external_account_authorized_user.py

```diff
@@ -321,6 +321,31 @@ def _build_trust_boundary_lookup_url(self):
             universe_domain=self._universe_domain, pool_id=pool_id
         )
 
+    def revoke(self, request):
```

Comment:
This looks great. One suggestion for improving the robustness :

After a successful revocation call to the STS client, should we also clear the local state of the credential? For example, by setting self._refresh_token = None and self.token = None.

This would make the credential object's state (valid and can_refresh properties) immediately consistent with the revoked status on the server and would prevent any future attempts to use or refresh this credential object. What do you think?